tests/lib/rules/object-curly-spacing.js

```diff
@@ -120,6 +125,12 @@ ruleTester.run("object-curly-spacing", rule, {
 
         // never - empty object
         { code: "var foo = {};", options: ["never"] },
+        { code: "var foo = {};", options: ["never", { spaceInEmptyObject: "never" }] },
+        { code: "var foo = { };", options: ["never", { spaceInEmptyObject: "always" }] },
+        { code: "var foo = {/* this is a comment */};", options: ["never", { spaceInEmptyObject: "always" }] },
```

Comment:
I think this case will currently report an error with `spaceInEmptyObject: "never"`, but I don't think that's desirable -- the rule shouldn't forbid people from putting comments in objects. Instead, I would expect the rule to enforce or forbid spaces *around* the comment in this case, depending on either the main option or the `spaceInEmptyObject` option.

Reply:
Are you saying it *should* report an error? I don't think the ~~"spaceInEmptyObject~~ "emptyObjects" option should have any bearing on an object that contains anything other than whitespace. 

Instead, we need yet another option for "commentsInObjects".

Reply:
I'm saying that the following code *should not* report an error, but it does report an error with the current implementation:

```js
/* eslint object-curly-spacing: [error, never, { spaceInEmptyObject: never }] */

var foo = {/* this is a comment */};
```

The issue with the current behavior is that `spaceInEmptyObject: "never"` is unsatisfiable for objects with a comment and no properties, because the rule still measures the offset between the `{` and the `}` rather than measuring the offset between the `{` and the start of the comment.

I don't feel strongly about how cases like the following should be treated:

```js
/* eslint object-curly-spacing: [error, always, { spaceInEmptyObject: never }] */

var foo = {/* this is a comment */};
```

I think I would want an error to be reported since there is no space between the `{` and the `/*`, and the object isn't "empty", but I'm open to being convinced otherwise.

Reply:
Okay, yes, that should not report an error.